server/prisma/schema.prisma

```diff
@@ -128,61 +118,68 @@ model events {
-  orders        orders          @relation(fields: [orderid_fk], references: [orderid], onDelete: Cascade, onUpdate: NoAction, map: "orderitems_orderid_fkey")
-  seasontickets seasontickets[]
-  singletickets singletickets[]
+}
+
+model order_ticketitems {
+  id         Int          @id @default(autoincrement())
+  orderid_fk Int
+  price      Decimal      @db.Money
+  order      orders       @relation(fields: [orderid_fk], references: [orderid], onDelete: Cascade, onUpdate: Cascade)
+  ticketitem ticketitems?
+  refund     refunditems?
+}
+
+/// supertable for sub ticket items tables (subscription, order)
+model ticketitems {
+  id                     Int                @id @default(autoincrement())
+  order_ticketitemid_fk  Int                @unique
+  ticketrestrictionid_fk Int
+  donated                Boolean            @default(false)
+  redeemed               DateTime?          @db.Timestamp(3)
+  ticketrestriction      ticketrestrictions @relation(fields: [ticketrestrictionid_fk], references: [ticketrestrictionsid], onDelete: Restrict, onUpdate: Cascade)
+  order_ticketitem       order_ticketitems  @relation(fields: [order_ticketitemid_fk], references: [id], onDelete: Cascade, onUpdate: Cascade)
 }
 
 /// a customers order
 /// refund and payment intents are the uuids from stripe when a refund or payment is made
 model orders {
-  orderid           Int          @id @default(autoincrement())
+  orderid           Int                 @id @default(autoincrement())
+  orderdatetime     DateTime            @default(now()) @db.Timestamp(3)
   contactid_fk      Int
-  orderdate         Int
-  ordertime         DateTime?    @db.Timetz(6)
   discountid_fk     Int?
-  payment_intent    String?      @db.VarChar(255)
-  checkout_sessions String?      @db.VarChar(255)
-  refund_intent     String?      @db.VarChar(255)
-  ordertotal        Decimal?     @db.Money
-  orderitems        orderitems[]
-  contacts          contacts     @relation(fields: [contactid_fk], references: [contactid], onDelete: NoAction, onUpdate: NoAction, map: "orders_contactid_fkey")
-  discounts         discounts?   @relation(fields: [discountid_fk], references: [discountid], onDelete: NoAction, onUpdate: NoAction, map: "orders_discountid_fkey")
-  date              date         @relation(fields: [orderdate], references: [dateid], onDelete: NoAction, onUpdate: NoAction)
+  payment_intent    String?             @unique @db.VarChar(255)
+  checkout_sessions String?             @unique @db.VarChar(255)
+  contacts          contacts            @relation(fields: [contactid_fk], references: [contactid], onDelete: Restrict, onUpdate: Cascade, map: "orders_contactid_fkey")
+  discounts         discounts?          @relation(fields: [discountid_fk], references: [discountid], onDelete: Restrict, onUpdate: Cascade, map: "orders_discountid_fkey")
+  donation          donations?
+  refunds           refunds[]
+  order_ticketitems order_ticketitems[]
+}
```

Comment:
I think we would still want to include the `ordertotal` in the order instead of having to fetch all the `orderitems` and check/apply a discount code to get the total.

Reply:
I added an ordersubtotal and discounttotal.